ci/cloudbuild/convert-to-branch-triggers.sh

```diff
@@ -31,48 +27,55 @@ source module ci/lib/io.sh
 
 function print_usage() {
   # Extracts the usage from the file comment starting at line 17.
-  sed -n '17,/^$/s/^# \?//p' "${PROGRAM_PATH}"
+  cat <<_EOM_
+Usage: ${PROGRAM_NAME} [options]
+  Options:
+    -h|--help          Print this help message
+    --branch=<name>    Use name instead of the current branch
+_EOM_
 }
 
 function convert_triggers() {
   local branch="$1"
   local name_prefix="${branch//./-}"
   shift
   for file in "$@"; do
-    sed \
+    sed -i \
       -e "s/^name: /name: ${name_prefix}-/" \
-      -e "s/branch: .*/branch: ${branch}/" "${file}" |
-      sponge "${file}"
+      -e "s/branch: .*/branch: ${branch}/" "${file}"
   done
 }
 export -f convert_triggers
 
 # Use getopt to parse and normalize all the args.
 PARSED="$(getopt -a \
   --options="h" \
-  --longoptions="help" \
+  --longoptions="help,branch:" \
   --name="${PROGRAM_NAME}" \
   -- "$@")"
 eval set -- "${PARSED}"
 
+BRANCH=""
 while true; do
   case "$1" in
     -h | --help)
       print_usage
       exit 0
       ;;
+    --branch)
+      BRANCH="$2"
+      shift 2
+      ;;
     --)
       shift
       break
       ;;
   esac
 done
 
-if [[ $# -lt 1 ]]; then
-  print_usage
-  exit 1
+if [[ -z "${BRANCH}" ]]; then
+  BRANCH="$(git branch --show-current)"
```

Comment:
Is the current branch really what we're looking for?  For example, in your current context I imagine you want to get "main" and not "ci-triggers-are-branch-specific" (and, say, "v1.42.x" if you were branched from that).

Reply:
That is why I added `--branch`. If I am running in `v1.42.x` I do not want `main`.  If I am running in `fix-v1.42.x-triggers` I do not want that branch (I want `v1.42.x` instead).

Reply:
Right, but I figured there must be some way to discover the name you really want from where you are running (not that I've put any thought into it beyond that).

Reply:
"the name you really want" can only be defined by convention.  If you are in a branch that looks like a release branch, then you probably want that branch.  If you are not in such a branch, you probably want the "parent" branch.  Leaving aside the fact that there may be multiple parent branches...  I found this horror:

https://stackoverflow.com/questions/3161204/how-to-find-the-nearest-parent-of-a-git-branch

I think that shell pipeline can be optimized (combining `grep` and `sed` calls), but frankly this is not that worthwhile.  I suspect most of the time we will just use the `--branch` option to be explicit.

Reply:
I removed the script. Let's discuss that in a separate PR and get the triggers committed first.